### ISSUE TITLE
awesome-typescript-loader error with 5.3.0-rc.6

### ISSUE THREAD

**Issue by kevinkuebler** [storybookjs/storybook#9285]
Trying to update from 5.2.5 to 5.3.0-rc.6 and started getting an error from awesome-typescript-loader. Going back to 5.2.5 works, but I get the error as soon as I update the storybook packages. Here are the relevant versions from my package.json:
```
"@storybook/addon-actions": "^5.3.0-rc.6",
"@storybook/addon-docs": "^5.3.0-rc.6",
"@storybook/addon-knobs": "^5.3.0-rc.6",
"@storybook/react": "^5.3.0-rc.6",
"@storybook/theming": "^5.3.0-rc.6",
"awesome-typescript-loader": "^5.2.1",
"react-docgen-typescript-loader": "^3.6.0",
"typescript": "^3.7.3",
"webpack": "^4.41.2",
"react": "^16.11.0",
"react-dom": "^16.11.0",
```
The error I'm getting when running Storybook 5.3.0-rc.6 looks like this:
```
ERROR in [at-loader] ./stories/addressField.tsx:27:33
    TS2306: File 'C:/Dev/Spry-React/stories/lockToggle.tsx' is not a module.
```
In this case, `addressField.tsx` contains a simple story in CSF format, and imports another component from the `lockToggle.tsx` file which is in the same directory, like this:
`import { LockToggleStory } from "./lockToggle";`

Our webpack.config for Storybook looks like this:
```
module.exports = ({config, mode}) => {
  config.stats = "errors-only";
  
  config.module.rules.push({
    test: /\.tsx?$/,
    use: ['awesome-typescript-loader', 'react-docgen-typescript-loader']
  });

  config.module.rules.push({
    test: /\.tsx$/,
    include: [/stories/],
    exclude: [/node_modules/],
    loader: require.resolve('@storybook/source-loader'),
    enforce: 'pre',
  });

  config.resolve = {
    extensions: ['.ts', '.js', '.tsx'],
    modules: [path.resolve(__dirname), "node_modules"],
    alias: {
      "spry-react": path.resolve(__dirname, "../src/")
    }
  };
  config.devtool = 'source-map';

  return config;
};
```
Any ideas why the new version of Storybook might be causing a problem with awesome-typescript-loader like this? I did notice today that the [awesome-typescript-loader repo](https://github.com/s-panferov/awesome-typescript-loader) has been archived by the owner. I'm wondering if the project has been abandoned by the owner? Has anyone seen any discussion or announcements about that? In the meantime I'm going to give ts-loader a shot and see if I have any better luck.

**Comment by kevinkuebler** [comment]
As a follow-up, I get the same error with ts-loader. So it definitely appears to be something that broke between Storybook 5.2 and 5.3. I may give the previous rc releases a shot to see if it happens in any of them too.

**Comment by kevinkuebler** [comment]
Some more information on this one ... I tried with both 5.3.0-rc.0 and 5.3.0-rc.8 and got the same results. So it appears this is something that broke between 5.2 and the first 5.3 RC release. What we're doing is probably considered "non-standared"/not best-practice, but goes back to how we were using Storybook before Storybook Docs was a thing. We would often write stories as a full React class component with state so that a user could interact with the component on the Storybook canvas like they would do in a real application.

I realize this isn't a full-blown executable sample of the problem, but maybe the following description of our configuration will help understand the problem. Lets say we have a component in our library named `AddressField` which is in `src/addressField.tsx`. We also have a HOC which produces "locked" versions of certain components in our library. We want to document both the normal `AddressField` and the `LockedAddressField` components, so we have a story for them in `stories/addressField.tsx`. That story file would look like this...

```
import * as React from "react";
import { AddressField, LockedAddressField } from "src/addressField";
import { LockedFieldStory } from "./lockedFieldStory";

export const simple = () => <AddressField />;
export const lockedAddressField = () => <LockedAddressFieldStory />;

class LockedAddressFieldStory extends React.Component<{}, State> {
  //Leaving out the constructor and other irrelevant details. The problem is in the render function we're using another React component that we imported. This is being used much like a decorator (we didn't know about decorators when we first wrote these stories - or maybe they didn't even exist yet, not sure).
  render() {
    return (
      <LockedFieldStory prop1={"foo"} prop2={"bar"}>
        <LockedAddressField />
      </LockedFieldStory>
    );
  }
}
```

That `LockedFieldStory` component is written as a regular React class component, with a couple of props (in addition to `children`) that it can take. This import is what appears to cause the error in 5.3 that I described in the original post.

I can convert this to use a decorator function instead, and that works (although I don't know if I can replicate the same functionality with a decorator that I could get with a component in terms of passing it some additional props?). But regardless it seems like this scenario _should_ work, correct? It does at least work currently in 5.2.5. If this isn't enough information to diagnose the issue, I can try to create a full running sample.

**Comment by shilman** [comment]
Hi @kevinkuebler I can't think of anything that might cause the break off the top of my head. @gaetanmaisse @kroeder @ndelangen @mrmckeb any ideas?

Best step forward would be to create a minimal repro and hopefully we can get to the bottom of it. Thanks for raising the issue!

**Comment by kevinkuebler** [comment]
Hi @shilman, yeah I figured you guys might need to see an actual repro to be able to investigate this one. 😄  Fortunately, I was able to simplify our configuration down to an example with just one simple component, and the use of the "wrapper" story component as I described, which demonstrates the issue. Attached is a simple repro using 5.3-rc9 which should show the error.

Ultimately, I'll probably end up converting our code to use decorators instead of doing this, but if this helps fix a bug that would be great and would allow me to test out 5.3 before I get them all converted. Thanks!
[Storybook9285Repro.zip](https://github.com/storybookjs/storybook/files/4024606/Storybook9285Repro.zip)
